### ISSUE TITLE
Ability to sync TicGit-ng with Github Issues

### ISSUE THREAD

**Issue by jeffWelling** [jeffWelling/ticgit#8]
Sync with Github Issues! That would be the shiznizzle!!!
Also, other bug trackers, but they can wait because they taste like cheeze and cardboard.
/jk about the cardboard-y cheesy taste ;)

Do you use another bug tracker that you would like to see TicGit-ng synchronize with? Start hacking! Or, if your not a developer, post a comment here with the bug tracker you would like to be able to synchronize with, and we'll light some candles and wave our magic programming stick and see what happens. :)

**Comment by heoa** [comment]
Are you planning to hard-code things with Github? 

-Please, do not do that rather create a scheme that all content-hosts can use. If someone use Github, he needs to put some plugin -file to some folder to get the support or enable some setting.

I cannot yet understand the problem here but personally I see no reason to waste time to read Github's API, rather create a generic tool to manage all kind of content-hosts -- and create clear goal like Version 1 (XYZ features) and Version 2 (ZYX features, not compatible with 1). 

Now the goal I think should be to outline what we want with such generic tools:
1. permission management (Gitolite)
2. ticket -management: versions, milestones (TicGit)
3. perhaps some features inspired by [1] (just novel idea, perhaps not needed)

Look since we do not know yet what users would like to have, I think a good way would be create everything pluggable. 

Some projects needs central tickets -management (Gitolite/TicGit), some projects may need central RSS -news, some projects may need specific permission -management (aka Gitolite), --.

Now too much stuff but some idea must be casted on how to make mixings: integration or pluggable or some other technique?

P.s. I think it is enough to have a timeline for different repos that gathers RSS feeds (from each repo, not sure how to get them but should not be hard), rather than create some side-effect -hack with submodules for the ticgit -branches. Anyway, my current idea that may change, thinking.

[1] https://www.flowdock.com/tour

**Comment by jeffWelling** [comment]
No, the synchronization module is being built so that all you have to do is write a (hopefully simple) read write interface for your bug tracker of choice and drop it into the the lib/ticgit-ng/sync folder (I think that's the location...) and run TicGit-ng and it should work.  Then ideally, if it works as you'd like it to, you could email me the file or submit a pull request and I could accept it, thus giving everyone who uses ticgit integration with the bug tracker that you use.

_I'm_ a user, and I know what _I_ want from TicGit, that's been good enough for me so far.  Not to be arrogant, just saying, it's not like there's a ton of users out there expressing an opinion about which way they think TicGit should be going (besides you and I), so I don't have a lot to go on.

Doesn't the idea of centralized tickets intrinsically imply that people regularly make use of more than one bug tracker?  

But anyway, yes, the sync-ing module is [already] written to support plug-in support for each and every bug tracker you'd care to add a plugin for, and I intent to include a couple default plug-ins as examples of how to do it.
